node/src/components/block_proposer.rs

```diff
@@ -263,9 +239,8 @@ impl BlockProposer {
     ) -> HashSet<DeployHash> {
         let past_deploys = self
             .state
-            .finalized
-            .values()
-            .flat_map(|deploys| deploys.keys())
+            .finalized_deploys
+            .keys()
```

Comment:
Now that `finalized_deploys` is a `HashMap`, lookup in it would be fast.
Instead of merging the collections, can we just filter by checking membership in both separately below?